internal/hcp/internal/controllers/link/controller.go

```diff
@@ -52,19 +112,70 @@ func (r *linkReconciler) Reconcile(ctx context.Context, rt controller.Runtime, r
 		return err
 	}
 
+	// Validation - Ensure V2 Resource APIs are not enabled unless hcpAllowV2ResourceApis flag is set
 	var newStatus *pbresource.Status
 	if r.resourceApisEnabled && !r.hcpAllowV2ResourceApis {
 		newStatus = &pbresource.Status{
 			ObservedGeneration: res.Generation,
 			Conditions:         []*pbresource.Condition{ConditionDisabled},
 		}
-	} else {
-		newStatus = &pbresource.Status{
-			ObservedGeneration: res.Generation,
-			Conditions:         []*pbresource.Condition{ConditionLinked(link.ResourceId)},
+		if resource.EqualStatus(res.Status[StatusKey], newStatus, false) {
+			return nil
+		}
+		_, err = rt.Client.WriteStatus(ctx, &pbresource.WriteStatusRequest{
+			Id:     res.Id,
+			Key:    StatusKey,
+			Status: newStatus,
+		})
+		if err != nil {
+			return err
+		}
```

Comment:
I think there needs to be a `return nil` here if the status write was successful so that we don't continue with the rest of the reconciliation.

Reply:
And then related to this, it looks like the disabled test has the mock client expecting to fetch the cluster from GNM, which I don't think should be the case.

Reply:
Thanks for pointing this out — the disabled test was wrong and hiding this bug. Since I wasn't doing `return nil` appropriately, the reconcile function continued on after writing the error status and tried to call `GetCluster`. Should be fixed now though!